src/structures/NewsChannel.js

```diff
@@ -13,6 +14,25 @@ class NewsChannel extends TextChannel {
     // News channels don't have a rate limit per user, remove it
     this.rateLimitPerUser = undefined;
   }
+
+  /**
+   * Adds the target to this channel's followers.
+   * @param {GuildChannelResolvable} channel The channel where the webhook should be created
+   * @param {string} [reason] Reason for creating the webhook
+   * @returns {Promise<NewsChannel>}
+   * @example
+   * if (channel.type === 'news') {
+   *   channel.addFollower('222197033908436994', 'Important announcements')
+   *     .then(() => console.log('Added follower'))
+   *     .catch(console.error);
+   * }
+   */
+  async addFollower(channel, reason) {
+    channel = this.client.channels.resolve(channel);
+    if (!channel || channel.type !== 'text') throw new Error('GUILD_TEXT_CHANNEL_RESOLVE');
```

Comment:
I find a couple of issues with this implementation:

- What if the channel is not cached? i.g. it's in another shard
- What if Discord adds more text-based channels and adds support for followers? We would need to change the message error, which is `semver: major`, and we can't afford that for a feature that is `semver: minor`.

Reply:
Any ideas on how to improve it? `resolve()` could be changed to `resolveID()` to prevent issues with caching but I don't like giving the user a `DiscordAPIError` unless really necessary

Reply:
I think we should stick with resolveID and not type check, that's on the user to check and it feels like the proposal of adding permissions check before API calls, which were denied multiple times before.

What are your thoughts? @vladfrangu @iCrawl

Reply:
I think I'd just reference what I did in #4285 - use `resolveID` and check that it does actually resolve to *something*. Once you've got a parameter, if its an invalid one the API can error.